Sources/RSDKUtilsMain/Lockable.swift

```diff
@@ -0,0 +1,104 @@
+import Foundation
+
+/// Protocol to mark object that have resources that can be thread locked
+public protocol Lockable {
+    var resourcesToLock: [LockableResource] { get }
+}
+
+public protocol LockableResource {
+
+    /// Lock resource for caller's thread use
+    func lock()
+
+    /// Unlock resource. Access from other threads will be resumed
+    func unlock()
+}
+
+/// Object-wrapper that conforms to LockableResource protocol.
+/// Used to control getter and setter of given resource. Can be used to create atomic transaction.
+///
+/// When lock() has been called on some thread, only that thread will be able to access the resource.
+/// Other threads will synchronously wait for unlock() call to continue.
+/// - Note: get() and set() are not thread-safe by itself
+public class LockableObject<T>: LockableResource {
```

Comment:
This has already been reviewed in other PRs yeah?

Reply:
Yes